python/astro_metadata_translator/file_helpers.py

```diff
@@ -0,0 +1,248 @@
+            error is encountered.
+
+        Returns
+        -------
+        md : `dict`
+            The requested header. `None` if it could not be read and
+            ``can_raise`` is `False`.
+        """
+        try:
+            return readMetadata(file, hdu=hdu)
+        except lsst.afw.fits.FitsError as e:
+            if can_raise:
+                # Try to convert a basic fits error code
+                if "(104)" in str(e):
+                    raise FileNotFoundError(f"No such file or directory: {file}") from e
+                raise e
+            return None
+
+except ImportError:
+    from astropy.io import fits
+
+    def read_fits_metadata(file, hdu, can_raise=False):
+        """Read a FITS header using astropy.
+
+        Parameters
+        ----------
+        file : `str`
+            The file to read.
+        hdu : `int`
+            The header number to read.
+        can_raise : `bool`, optional
+            Indicate whether the function can raise and exception (default)
+            or should return `None` on error. Can still raise if an unexpected
+            error is encountered.
+
+        Returns
+        -------
+        md : `dict`
+            The requested header. `None` if it could not be read and
+            ``can_raise`` is `False`.
+        """
+        header = None
+        try:
+            with fits.open(file) as fits_file:
+                try:
+                    header = fits_file[hdu].header
+                except IndexError as e:
+                    if can_raise:
+                        raise e
+        except Exception as e:
+            if can_raise:
+                raise e
+        return header
+
+
+def find_files(files, regex):
+    """Find files for processing.
+
+    Parameters
+    ----------
+    files : iterable of `str`
+        The files or directories from which the headers are to be read.
+    regex : `str`
+        Regular expression string used to filter files when a directory is
+        scanned.
+    """
+    file_regex = re.compile(regex)
+    found_files = []
+
+    # Find all the files of interest
+    for file in files:
+        if os.path.isdir(file):
+            for root, dirs, files in os.walk(file):
+                for name in files:
+                    path = os.path.join(root, name)
+                    if os.path.isfile(path) and file_regex.search(name):
+                        found_files.append(path)
+        else:
+            found_files.append(file)
+
+    return found_files
+
+
+def read_basic_metadata_from_file(file, hdrnum, errstream=sys.stderr, can_raise=False):
+    """Read a raw header from a file, merging if necessary
+
+    Parameters
+    ----------
+    file : `str`
+        Name of file to read. Can be FITS or YAML. YAML must be a simple
+        top-level dict.
+    hdrnum : `int`
+        Header number to read. Only relevant for FITS. If greater than 1
+        it will be merged with the primary header.
```

Comment:
it's not defaulted here but comes from the click command. I think I defaulted it to 1 because there were files around the place that didn't work by default with no merging.  The YAML option here is mostly to support internal testing and I don't expect users to be using YAML at all.